src/mmw/apps/modeling/views.py

```diff
@@ -254,22 +256,29 @@ def start_mapshed(request, format=None):
 
 
 def _initiate_mapshed_job_chain(mapshed_input, job_id):
+    workers = choose_workers()
+    get_worker = lambda: random.choice(workers)
```

Comment:
This seems to be the most significant difference between the old version. Instead of making a query for a celery worker for each task in the job chain, you query the list of workers once for the whole job chain.

I'd be curious to see how much of an impact this change has by itself, without any of the other changes.